compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ModuleContext.java

```diff
@@ -0,0 +1,15 @@
+package io.ballerina.projects.internal;
+
+import io.ballerina.projects.ModuleDescriptor;
+
+/**
+ * Interface for {@code ModuleContext}.
+ *
+ * @since 2.0.0
+ */
+public interface ModuleContext {
```

Comment:
I have mixed feelings about this approach. Here we create an interface in the internal package, and the public `ModuleContext` class implements it. Feels like a hack. 

Can we create a class called `ModuleContextDataHolder` or something inside the internal package? Then we can create a new instance of this class for each `ModuleContext` and store it in `BLangPackage`.

Reply:
Added `ModuleContextDataHolder` and removed `ModuleContext` interface